src/FactsGatherer/SeekableStreamFactsGatherer.php

```diff
@@ -0,0 +1,117 @@
+<?php
+declare(strict_types = 1);
+
+namespace ZipStream\FactsGatherer;
+
+use ZipStream\File\DeflatedFileFacts;
+use ZipStream\File\DeflatedFileFactsInterface;
+use ZipStream\File\FileFacts;
+use ZipStream\File\FileFactsInterface;
+use ZipStream\File\FileInterface;
+use ZipStream\File\SeekableStreamFileInterface;
+
+/**
+ * Class SeekableStreamFactsGatherer
+ * @package ZipStream\FactsGatherer
+ */
+class SeekableStreamFactsGatherer implements DeflatedFactsGathererInterface
+{
+    /**
+     * @var int
+     */
+    private $blockSize;
+
+    /**
+     * @var resource
+     */
+    private $hashingContext;
+
+    /**
+     * SeekableStreamFactsGatherer constructor.
+     * @param int $blockSize
+     */
+    public function __construct(
+        int $blockSize = 1048576 // 1 MB
+    ) {
+
+        $this->blockSize = $blockSize;
+        $this->hashingContext = hash_init('crc32b');
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function supports(FileInterface $file): bool
+    {
+        return $file instanceof SeekableStreamFileInterface;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param SeekableStreamFileInterface $file
+     */
+    public function gatherFacts(FileInterface $file): FileFactsInterface
```

Comment:
I think all of those things could be collected in one iteration of the stream. I'll adapt that.